src/ucp/wireup/wireup_cm.c

```diff
@@ -1143,10 +1200,17 @@ ucp_ep_cm_server_create_connected(ucp_worker_h worker, unsigned ep_init_flags,
         goto err_destroy_ep;
     }
 
+    if (ep_init_flags & UCP_EP_INIT_CREATE_AM_LANE_ONLY) {
+        address_mode = UCP_WIREUP_SA_DATA_AM_ONLY;
+    } else {
+        address_mode = UCP_WIREUP_SA_DATA_CM_ADDR;
+    }
```

Comment:
add helper function for this?
or just pass ep_init_flags to ucp_ep_cm_connect_server_lane()->init_priv_data()  and do this if/else in init_priv_data